data/json/items/ammo/300.json

```diff
@@ -32,7 +32,11 @@
       "damage": 0.9,
       "dispersion": 1.1
     },
-    "extend": { "effects": [ "RECYCLED" ] },
-    "delete": { "effects": [ "NEVER_MISFIRES" ] }
+    "extend": {
+      "effects": [ "RECYCLED" ]
```

Comment:
This should be able to stay on one line

Reply:
There's no way to do that without backtracking.

Reply:
Requiring the formatter to be one-pass is a huge restriction that may make it unusable.

Reply:
Aggressive inlining is a nice-to-have at best and an antipattern at worst.

As that is currently the sole reason to make the formatter significantly more complex, I'm not inclined to invest the time to implement it.  If you want to write a backtracking formatter that does smart line wrapping and doesn't require external annotation, feel free.